Source/MQTTnet/Implementations/CrossPlatformSocket.cs

```diff
@@ -119,9 +119,29 @@ public async Task ConnectAsync(string host, int port, CancellationToken cancella
 #if NET452 || NET461
                     await Task.Factory.FromAsync(_socket.BeginConnect, _socket.EndConnect, host, port, null).ConfigureAwait(false);
 #else
-                    await _socket.ConnectAsync(host, port).ConfigureAwait(false);
+                    Task socketAwait = Task.Run(() =>
+                    {
+                        try
+                        {
+                            _socket.Connect(host, port);
+                        }
+                        catch (SocketException)
+                        {
+                        }
+                    });
+
+                    if (socketAwait.Wait(TimeSpan.FromSeconds(10)))
```

Comment:
This approach will lead to a sync call which will block the calling thread. So it will free UIs etc. Is there another way to ding this without blocking?

Reply:
I'll do some pondering. FWIW, this was blocking the Xamarin UI completely when the socket call would hang, but I can see your point that not blocking at all would be preferable.

Reply:
I made some changes. Let me know your thoughts. I actually have to apologize -- I accidentally PR'd a version of the code I was using when toying around, trying to explore what was up. This is the one I had intended to PR. That's why I was initially confused about your comment.